test/controllers/sections/items/ItemGeographicalIndicationChoiceControllerSpec.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sections.items
+
+import base.SpecBase
+import forms.sections.items.ItemGeographicalIndicationChoiceFormProvider
+import mocks.services.MockUserAnswersService
+import models.GoodsTypeModel.Beer
+import models.NormalMode
+import models.sections.items.ItemGeographicalIndicationType.{ProtectedDesignationOfOrigin, ProtectedGeographicalIndication}
+import navigation.FakeNavigators.FakeItemsNavigator
+import navigation.ItemsNavigator
+import pages.sections.items.{ItemExciseProductCodePage, ItemGeographicalIndicationChoicePage}
+import play.api.inject.bind
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.UserAnswersService
+import views.html.sections.items.ItemGeographicalIndicationChoiceView
+
+import scala.concurrent.Future
+
+class ItemGeographicalIndicationChoiceControllerSpec extends SpecBase with MockUserAnswersService {
+
+  val formProvider = new ItemGeographicalIndicationChoiceFormProvider()
+  val form = formProvider()
+  val baseUserAnswers = emptyUserAnswers.set(ItemExciseProductCodePage(testIndex1), "B000")
+
+  val action = controllers.sections.items.routes.ItemGeographicalIndicationChoiceController.onSubmit(testErn, testDraftId, testIndex1, NormalMode)
+
+  lazy val itemGeographicalIndicationChoiceRoute = routes.ItemGeographicalIndicationChoiceController.onPageLoad(testErn, testDraftId, testIndex1, NormalMode).url
+
+  "ItemGeographicalIndicationChoice Controller" - {
+
+    "must return OK and the correct view for a GET" in {
```

Comment:
Add tests for when Idx is out of bounds, once that's added to the Controller methods

Reply:
Sorted